src/DashboardService.Business/Priority/CreatePriorityCommand.cs

```diff
@@ -1,23 +1,62 @@
+using FluentValidation.Results;
 using LT.DigitalOffice.DashboardService.Business.Priority.Interfaces;
 using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Mappers.Db.Interfaces;
 using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+using LT.DigitalOffice.DashboardService.Validation.Priority.Interfaces;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
 using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Http;
 using System;
+using System.Linq;
+using System.Net;
 using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.DashboardService.Business.Priority;
 
 public class CreatePriorityCommand : ICreatePriorityCommand
 {
   private readonly IPriorityRepository _priorityRepository;
+  private readonly IAccessValidator _accessValidator;
+  private readonly ICreatePriorityRequestValidator _validator;
+  private readonly IResponseCreator _responseCreator;
+  private readonly IDbPriorityMapper _priorityMapper;
+  private readonly IHttpContextAccessor _httpContextAccessor;
 
-  public CreatePriorityCommand(IPriorityRepository priorityRepository)
+  public CreatePriorityCommand(
+    IPriorityRepository priorityRepository,
+    IAccessValidator accessValidator,
+    ICreatePriorityRequestValidator validator,
+    IResponseCreator responseCreator,
+    IDbPriorityMapper priorityMapper,
+    IHttpContextAccessor httpContextAccessor)
   {
     _priorityRepository = priorityRepository;
+    _accessValidator = accessValidator;
+    _validator = validator;
+    _responseCreator = responseCreator;
+    _priorityMapper = priorityMapper;
+    _httpContextAccessor = httpContextAccessor;
   }
-  
-  public Task<OperationResultResponse<Guid?>> ExecuteAsync(CreatePriorityRequest request)
+
+  public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreatePriorityRequest request)
   {
-    throw new NotImplementedException();
+    if (!await _accessValidator.IsAdminAsync())
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.Forbidden);
+    }
+    
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(vf => vf.ErrorMessage).ToList());
+    }
+
+    _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+    return new() {Body = await _priorityRepository.CreateAsync(_priorityMapper.Map(request))};
```

Comment:
split. no need for 1-liners

Reply:
You don't check repository result. In case of error you will return 201 - null anyway

Reply:
createAsync method contains only 3 lines 
![image](https://github.com/LT-Students/DashboardService/assets/42372955/677754d1-85af-456f-bad1-cd7f627009ae)
and there is no return any errors. if we pass validation - we just create entity in DB.

Reply:
> split. no need for 1-liners

++